modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/model/listener/FragmentEntryLinkModelListener.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.fragment.internal.model.listener;
```

Comment:
I'm just wondering where should this be placed... is the responsibility of `fragment-impl` or should this be a separate piece that adds react support for fragments? I'm assuming this mode-listener could be anywhere so maybe it makes more sense somewhere else... just floating the idea

Reply:
Note that this could also be implemented in the LocalService itself.

Also note that, as this is done now, it has a performance problem when change propagation is activated because that will trigger one NPMRegistryUpdate (which is a costly operation) per FragmentEntryLink. 

The ideal way of doing it would be using the same NPMRegistryUpdate for all the propagations, perhaps by storing it in a ThreadLocal in the ModelListener or a similar technique.

Reply:
Yes, but does it need to live inside `fragment-impl`? or could this be a separate thing? trying to make sure we are setting the proper boundaries here and not coupling concepts.

Reply:
You can put it outside fragment-impl, but then the fragment renderer may break if a React fragment is imported (as it is assuming this thing is in place). 

So I guess it makes sense to put it here...